### ISSUE TITLE
about concat

### ISSUE THREAD

**Issue by chencodeX** [automan000/Convolution_LSTM_PyTorch#4]
https://arxiv.org/pdf/1506.04214.pdf
in this parper
![image](https://user-images.githubusercontent.com/14937564/39239144-9e692f2a-48b2-11e8-91ce-dd9754a7d8b5.png)
the LSTM compute i(t),f(t),o(t) is use x(t),h(t-1) and c(t-1). but in your code convolution_lstm.py line 24:
combined = torch.cat((input, h), dim=1)

you just use x and h, why?

**Comment by automan000** [comment]
Hi Chen,
I think you might be right. I forget why I did this when I wrote the code.
I have checked several implementations. Many of them make the same decision.
I will do some further investigations. Once finished, I will let you know.
Thanks anyway.

**Comment by chencodeX** [comment]
I have some of my own opinions, which may be to save code complexity and the number of parameters

**Comment by Jackie-Chou** [comment]
@automan000 Guess you have referenced some implementations of **LSTM** instead of **convLSTM**, note the difference of formulas of them .
**LSTM**
![image](https://user-images.githubusercontent.com/20188057/39963557-3975ae48-56a1-11e8-9762-d3f8a831cd9f.png)
**convLSTM**
![image](https://user-images.githubusercontent.com/20188057/39963558-454f81f8-56a1-11e8-85b2-b6967672182d.png)
The raw LSTM really doesn't involve c for computing the gate while convLSTM does.

**Comment by automan000** [comment]
@Jackie-Chou Actually I am referring the implementations such as [BasicConvLSTMCell](https://github.com/loliverhennigh/Convolutional-LSTM-in-Tensorflow/blob/master/BasicConvLSTMCell.py)
But your words are very convincing. I will update the code ASAP. Thanks a lot.

**Comment by automan000** [comment]
Bug fixed. Close this issue.
